### PR TITLE
Added custom 500 page

### LINKED ISSUE #487: General 500 error page
Should include a link to mitx-support@mit.edu, with the subject set to the error message.

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
closes #487 

#### What's this PR do?
Adds a custom 500 page

#### Where should the reviewer start?
`ui.views` and then '500.html'

#### How should this be manually tested?
to see the the 500 template, just go to `/500/`.
To actually test that a 500 is returned for a error, you need to:
  - set the debug to False in your `docker-compose.yml`
  - run `node node_modules/webpack/bin/webpack.js --config webpack.config.prod.js --bail` to build the static assets
  - run `docker-compose run web ./manage.py collectstatic` 
  - raise any exception in a view like `ui.views.dashboard`

### CODE REVIEW

File: ui/views_test.py [hunk 1]
```diff
@@ -252,6 +252,29 @@ def test_404_error_context_logged_out(self):
             assert response.context['authenticated'] is False
             assert response.context['name'] == ""
 
+    def test_500_error_context_logged_in(self):
```
Comment (noisecapella) [inline ui/views_test.py]: Can we add a test that causes a 500 error and verifies the page is shown, maybe by mocking something to raise an exception?

**Comment by noisecapella** [comment]
Functionality looks good

**Comment by amir-qayyum-khan** [comment]
@giocalitri  i am curious, how this 500 error is linked with django crash with 500 status code? ofcourse we disable default 500 error by setting this:

```set the debug to False in your docker-compose.yml```

**Comment by pdpinch** [comment]
@giocalitri can you add a screenshot of the error message to the issue #487 and @roberthouse54 to take a look at it?

File: ui/views_test.py (hunk 1 above)
Comment (giocalitri) [inline ui/views_test.py]: after trying few approaches, I realized that it is not currently possible. 
See https://code.djangoproject.com/ticket/18707
and it looks they are non going to implement it any time soon

**Comment by noisecapella** [comment]
Looks good to me :+1:

**Comment by giocalitri** [comment]
@amir-qayyum-khan I'm not sure what you mean.

Django by default has an empty error page. What this PR is doing is overriding that view.

**Pull request merged**